src/plugins/profiling/server/plugin.ts

```diff
@@ -9,45 +9,37 @@ import { PluginInitializerContext, CoreSetup, CoreStart, Plugin, Logger } from '
 
 import type { DataRequestHandlerContext } from '../../data/server';
 
-import {
-  ProfilingPluginSetup,
-  ProfilingPluginStart,
-  ProfilingPluginSetupDeps,
-  ProfilingPluginStartDeps,
-} from './types';
-import { mySearchStrategyProvider } from './my_strategy';
+import { ProfilingPluginSetupDeps, ProfilingPluginStartDeps } from './types';
 import { registerRoutes } from './routes';
 
 export class ProfilingPlugin
-  implements
-    Plugin<
-      ProfilingPluginSetup,
-      ProfilingPluginStart,
-      ProfilingPluginSetupDeps,
-      ProfilingPluginStartDeps
-    >
+  implements Plugin<void, void, ProfilingPluginSetupDeps, ProfilingPluginStartDeps>
 {
   private readonly logger: Logger;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
   }
 
-  public setup(core: CoreSetup<ProfilingPluginStartDeps>, deps: ProfilingPluginSetupDeps) {
+  public setup(core: CoreSetup<ProfilingPluginStartDeps>, { data }: ProfilingPluginSetupDeps) {
     this.logger.debug('profiling: Setup');
+    // TODO we should create a query here using "data".
+    // We should ensure there are profiling data in the expected indices
```

Comment:
IMO we should check at setup time that the indices exist and at query time that there is data available (and tell the user if there isn't).